### PR TITLE
Allow custom code

### PR DESCRIPTION
**Pull request by ruhley**
Uses swagger-js-codegen's getCustomCode method for more flexibility

### CODE REVIEW

**Comment by wcandillon** [comment]
@ruhley Awesome! Can you fix the build? I will merge it then.

**Comment by ruhley** [comment]
I have fixed it up. There is a change that would break some things. I have changed the ```angularjs``` boolean option to ```type``` with the options ```'angular'```, ```'node'``` and ```'custom'``` with custom being the default. I can change this to something else if you like.

File: tasks/swagger-js-codegen.js [hunk 1]
```diff
@@ -24,8 +24,16 @@ module.exports = function (grunt) {
                     if(error || response.statusCode !== 200) {
                         deferred.reject('Error while fetching ' + api.swagger + ': ' + (error || body));
                     } else {
-                        var swagger = JSON.parse(body);
-                        var source = api.angularjs === true ? CodeGen.getAngularCode({ moduleName: api.moduleName, className: api.className, swagger: swagger }) : CodeGen.getNodeCode({ className: api.className, swagger: swagger });
+                        var swagger = JSON.parse(body),
+                            source = null;
+
+                        if (api.type === 'angular') {
```
Comment (wcandillon) [inline tasks/swagger-js-codegen.js]: @ruhley can you do:
```javascript
if(api.type === 'angular' || api.angularjs === true) {
```
?

**Comment by wcandillon** [comment]
The change below should make it backward compatible right?
```javascript
if(api.type === 'angular' || api.angularjs === true) {
```

**Comment by ruhley** [comment]
The old system defaulted to node, but this still defaults to custom. I can change it to
```javascript
if (api.angularjs === true) {
    // angularjs codegen
} else if (api.custom === true) {
    // custom codegen
} else {
    // node codegen
}
```

**Comment by wcandillon** [comment]
@ruhley It make sense to default to node.

**Comment by ruhley** [comment]
Changed to my example just above.

**Pull request merged**